sdk/needsauth/helpers.go

```diff
@@ -45,27 +58,79 @@ func Always() sdk.NeedsAuthentication {
 	}
 }
 
-// NotForArgs returns a NeedsAuthentication rule to not require authentication when
-// certain command-line args or flags are present.
-func NotForArgs(argsToSkip ...string) sdk.NeedsAuthentication {
+// NotForExactArgs returns a NeedsAuthentication rule to opt out of authentication when
+// the command-line args are an exact match with the passed in args.
+func NotForExactArgs(argsToSkip ...string) sdk.NeedsAuthentication {
 	return func(in sdk.NeedsAuthenticationInput) bool {
-		for _, commandArg := range in.CommandArgs {
-			for _, ignoreArg := range argsToSkip {
-				if commandArg == ignoreArg {
-					return false
+		if len(in.CommandArgs) != len(argsToSkip) {
+			return true
+		}
+
+		for i, commandArg := range in.CommandArgs {
+			if commandArg != argsToSkip[i] {
+				return true
+			}
+		}
+
+		return false
+	}
+}
+
+// NotWhenContainsArgs returns a NeedsAuthentication rule to not require authentication when
+// the exact sequence of argsToSkip is present somewhere in the command-line args.
+func NotWhenContainsArgs(argsToSkip ...string) sdk.NeedsAuthentication {
```

Comment:
```suggestion
func NotWhenContainsArgsSequence(argsToSkip []string) sdk.NeedsAuthentication {
```
Should this be renamed by adding the `Sequence` keyword so to imply that a match is made for >2 terms only if they are found in the exact order they have been specified?

What would be the use case of calling `NotWhenContainsArgsSequence()`?

Reply:
Or how about:
```suggestion
func NotWhenContainsArgs(argsSequence ...string) sdk.NeedsAuthentication {
```

Reply:
Yes, this makes sense.